core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -976,6 +976,12 @@ class KafkaApis(val requestChannel: RequestChannel,
           else
             topicMetadata
         } else if (allowAutoTopicCreation && config.autoCreateTopicsEnable) {
+          val msg =  "Automatically creating topic: " + topic + " with " + config.numPartitions +
```

Comment:
@Lincong and I talked about this offline.

The motivation for this is as a preparatory step to turning off `auto.create` in a cluster in order to identify which applications _might_ be impacted.

Logging an auto-create event is fine by itself, but I don't think this patch is particularly useful for the said motivation.
Most users have some access logging mechanism (e.g., in the authorizer implementation) and capture this information. These metadata requests are likely immediately followed up by produce/consume so the set of principals that produce/consume is really the information you are after (as opposed to the principal that _happened_ to first create the topic).

Separately, this logging should probably live further down the stack - e.g., there could be a subsequent error (e.g., `TopicExists`) so logging this here is slightly misleading.